src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+package org.broadinstitute.hellbender.tools.spark.sv.evidence;
+
+import biz.k11i.xgboost.Predictor;
+import biz.k11i.xgboost.learner.ObjFunction;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.TextCigarCodec;
+import htsjdk.tribble.AbstractFeatureReader;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.*;
+import org.broadinstitute.hellbender.tools.spark.utils.IntHistogram;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import scala.Tuple2;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * A class that acts as a filter for BreakpointEvidence.
+ * Features are calculated according to evidence type, overlap information, mapping quality, etc.
+ * A trained classifier scores the probability the evidence overlaps a breakpoint interval, and passes evidence above
+ * the specified threshold.
+ */
+public final class XGBoostEvidenceFilter implements Iterator<BreakpointEvidence> {
+    static private final boolean useFastMathExp = true;
+    private static final List<String> defaultEvidenceTypeOrder = Arrays.asList(
+            "TemplateSizeAnomaly",
+            "MateUnmapped", "InterContigPair",
+            "SplitRead", "LargeIndel", "WeirdTemplateSize", "SameStrandPair", "OutiesPair"
```

Comment:
Yeah, I've had ExternalEvidence on my mind (part of the reason I put the override capability in) but I ignored for now because it hasn't showed up in evidence dumps. Basically I'd need to dump the evidence, train a new classifier, and update the evidence order. I don't think there's a reasonable way to tell a classifier to pretend it knows how to deal with a totally new evidence type, especially because that is the most important feature as ranked by the training algorithm.

Reply:
I think we should treat `ExternalEvidence` as a first class evidence type (that's on and part of the classifier by default) but to do that we first need to wait until there's a stable gCNV/GATK-SV pipeline and gCNV callset we can use for `ExternalEvidence`. If we want to get fancy we could provide two classifiers, one trained with external evidence and one without. What do you think?